doc/api/string_decoder.md

```diff
@@ -17,23 +23,50 @@ const euro = Buffer.from([0xE2, 0x82, 0xAC]);
 console.log(decoder.write(euro));
 ```
 
-## Class: StringDecoder
+When a `Buffer` instance is written to the `StringDecoder` instance, an
+internal buffer is used to ensure that the decoded string does not contain
+any incomplete multibyte characters. These are held in the buffer until the
+next call to `stringDecoder.write()` or until `stringDecoder.end()` is called.
+
+In the following example, the three UTF-8 encoded bytes of the European euro
+symbol are written over three separate operations:
+
+```js
+const StringDecoder = require('string_decoder').StringDecoder;
+const decoder = new StringDecoder('utf8');
+
+decoder.write(Buffer.from([0xE2]));
+decoder.write(Buffer.from([0x82]));
+console.log(decoder.write(Buffer.from([0xAC])));
+```
+
+## Class: new StringDecoder([encoding])
 <!-- YAML
 added: v0.1.99
 -->
 
-Accepts a single argument, `encoding` which defaults to `'utf8'`.
+* `encoding` {string} The character encoding the `StringDecoder` will use.
+  Defaults to `'utf8'`.
+
+Creates a new `StringDecoder` instance.
 
-### decoder.end()
+### stringDecoder.end()
```

Comment:
Actually `end()` can accept a `Buffer` just like `write()`. It's a convenience thing that calls `.write()` first before adding any replacement characters.